casper/src/main/scala/coop/rchain/casper/blocks/BlockReceiver.scala

```diff
@@ -261,6 +261,17 @@ object BlockReceiver {
               hasAllDeps = pendingRequests.isEmpty &&
                 block.justifications.forall(dag.contains)
 
+              // If replay was interrupted, block was stored but not validated or added to the DAG.
+              // Validation needs to be restarted for such blocks
+              unvalidatedParents <- block.justifications.filterA { hash =>
+                                     BlockStore[F].contains(hash) &&^ dag.contains(hash).pure.not
+                                   }
+              _ <- unvalidatedParents.traverse { hash =>
```

Comment:
How this change helps when node starts and receives a block which is already in the block store? It will not be added to the _incomingQueue_ so this method is not invoked. Am I missing something?
https://github.com/rchain/rchain/blob/efa33eb02b291268eb937c03fd7ec1dca90e574d/casper/src/main/scala/coop/rchain/casper/engine/NodeRunning.scala#L237-L259

Reply:
Yes, you are correct. Thanks! I didn't consider that case. I [tried](https://github.com/rchain/rchain/pull/3803/commits/c774f63777d1d8c56f4c274ad4bb363c72954e1d) to fix it.